lib/kubeclient/common.rb

```diff
@@ -291,6 +310,48 @@ def get_entities(entity_type, klass, resource_name, options = {})
       Kubeclient::Common::EntityList.new(entity_type, resource_version, collection)
     end
 
+    def get_entities_async(klass, resource_name, options = {})
+      if options[:as] == :raw
+        raise ArgumentError('Cannot pass block to get_entities when requesting options[:as] == raw')
+      end
+
+      params = {}
+      SEARCH_ARGUMENTS.each { |k, v| params[k] = options[v] if options[v] }
+
+      ns_prefix = build_namespace_prefix(options[:namespace])
+
+      resource_version = nil
+      entity_streamer = Json::Streamer::JsonStreamer.new
+      resource_version_streamer = Json::Streamer::JsonStreamer.new
+
+      resource_version_streamer.get(key: 'resourceVersion') do |value|
+        resource_version = value.to_i unless resource_version
+      end
+
+      # result['items'] might be nil due to https://github.com/kubernetes/kubernetes/issues/13096
+      entity_streamer.get(nesting_level: 2, yield_values: false) do |object|
```

Comment:
again, can this be narrowed to only 'items'?  if k8s adds another top-level array, it will confuse us.

Reply:
i tried doing this and discovered (what i believe is) a bug with `json-streamer`: https://github.com/thisismydesign/json-streamer/issues/7

though the bug possibly resides with `json-stream` itself. it may require forking `json-streamer` or re-implementing some of its features as mentioned above :(

Reply:
Seems that was fixed and you can bump the dependency?
Need not block this, if still in flux/untested, I'm fine with merging this and improving later.

Reply:
fixed!